source/engine/graphics/Graphics.cpp

```diff
@@ -192,32 +156,46 @@ namespace nc::graphics
         if (m_isMinimized) return false;
 
         // Gets the next image in the swapchain
-        GetNextImageIndex();
+        if(!m_swapchain->GetNextRenderReadyImageIndex(m_frameManager->CurrentFrameContext(), &m_imageIndex))
+        {
+            RecreateSwapchain(m_dimensions);
+        }
+
+        m_frameManager->Begin();
 
         return true;
     }
 
     // Executes the command buffer for the next swapchain image which writes to the image.
     // Then, returns the image written to to the swap chain for presentation.
-    // Note: All calls below are asynchronous fire-and-forget methods. A maximum of Device::MaxFramesInFlight sets of calls will be running at any given time.
-    // See Device.cpp for synchronization of these calls.
+    // Note: All calls below are asynchronous fire-and-forget methods. A maximum of MaxFramesInFlight sets of calls will be running at any given time.
     void Graphics::Draw(const PerFrameRenderState& state)
     {
         OPTICK_CATEGORY("Graphics::Draw", Optick::Category::Rendering);
         if (m_isMinimized) return;
 
-        m_renderer->Record(m_commands.get(), state, m_gpuAssetsStorage.get()->meshStorage, m_imageIndex);
+        m_renderer->Record(m_frameManager->CurrentFrameContext(), state, m_gpuAssetsStorage.get()->meshStorage, m_imageIndex);
 
         // Executes the command buffer to render to the image
-        RenderToImage(m_imageIndex);
+        OPTICK_CATEGORY("Graphics::RenderToImage", Optick::Category::Rendering);
```

Comment:
We should only have one optick tag per scope. We don't want 3 different profiling events that fall under the same function. I think (?) we're fine just profiling `Draw()` here, but if we really want to profile the 3 sections, this strongly suggests the function should be split up into 3 functions.